quix-frontend/client/src/lib/sql-autocomplete/adapter/trinoToJs.ts

```diff
@@ -0,0 +1,146 @@
+const enum StartingOptions {
+  Row = 'row(',
+  Map = 'map(',
+  Array = 'array(',
+  TimeStamp = 'timestamp(',
+}
+const enum SpecialCharacters {
+  OpenParenthesis = '(',
+  ClosedParenthesis = ')',
+  Comma = ',',
+  Space = ' ',
+}
+
+export function trinoToJs(trinoObjectAsString: string, index: number):  object|string {
+  if (trinoObjectAsString.startsWith(StartingOptions.Row , index)) {
+    return processRow(trinoObjectAsString, index + StartingOptions.Row.length);
+  }
+  if (trinoObjectAsString.startsWith(StartingOptions.Map, index )) {
+    return 'map';
+  }
+
+  if ( trinoObjectAsString.startsWith(StartingOptions.Array, index ) ) {
+    return 'array';
+  }
+    return trinoObjectAsString;
+}
+
+export function processRow(trinoObjectAsString: string, index: number): object {
+  const finalObject: object = {};
+  let key = "";
+  let value = "";
+  let objectToInsert;
+  let isFirstWord = true;
+  const lengthOfSubstring = 5;
+
+  while (index <= trinoObjectAsString.length) {
+    switch (trinoObjectAsString.charAt(index)) {
+      case SpecialCharacters.OpenParenthesis: {
+        const start = findBeginningOfWord(trinoObjectAsString, index);
+        const result = handleStartingOptions(trinoObjectAsString, start, index);
+        objectToInsert = result.objectToInsert;
+        finalObject[key] = result.type;
+        index = result.newIndex;
+        break;
+      }
+      case SpecialCharacters.ClosedParenthesis: {
+        validateKey(key, index);
+        finalObject[key] = objectToInsert || value;
+        return finalObject;
+      }
+      case SpecialCharacters.Comma: {
+        if (value === "") {
+          throw new Error(`Error at index: ${trinoObjectAsString.substring(index-lengthOfSubstring,index)}, type expected before comma`);
+        }
+        finalObject[key] = objectToInsert || value;
+        key = "";
+        value = "";
+        objectToInsert = undefined;
+        isFirstWord = true;
```

Comment:
this logic is also in line 30-34. consider create a builder function for it.

Reply:
i agree with you, but i think in this scenario it will be easier to understand if we keep it like this.
i think the flow is easier to follow like this

Reply:
it could be confusing to see same finction called in the begining of the function and in the middle of the loop